packages/ui/src/modals/UpdateMembershipModal/SignUpdateMembershipModal.tsx

```diff
@@ -11,33 +13,56 @@ import { useApi } from '../../hooks/useApi'
 import { useSignAndSendTransaction } from '../../hooks/useSignAndSendTransaction'
 import { BalanceInfoNarrow, InfoTitle, InfoValue, Row } from '../common'
 import { WaitModal } from '../WaitModal'
-import { UpdateMemberForm } from './UpdateMembershipFormModal'
+import { Nullable, UpdateMemberForm } from './types'
 
 interface SignProps {
   onClose: () => void
-  transactionParams: UpdateMemberForm
+  transactionParams: Nullable<UpdateMemberForm>
   onDone: (result: boolean, fee: BN) => void
   member: BaseMember
+  signer: Account
 }
 
-export const SignUpdateMembershipModal = ({ onClose, transactionParams, member, onDone }: SignProps) => {
-  const { api } = useApi()
-  const updateProfileTransaction = useMemo(
-    () =>
-      api?.tx?.members?.updateProfile(
-        member.id,
-        transactionParams.name as string,
-        transactionParams.handle as string,
-        transactionParams.avatarURI as string,
-        transactionParams.about as string
-      ),
-    [member.id]
-  )
+const hasEdits = (object: Record<string, any>, fields: string[]) => {
+  return fields.some((field) => !!object[field])
+}
+
+function createBatch(transactionParams: Nullable<UpdateMemberForm>, api: ApiRx | undefined, member: BaseMember) {
+  const hasProfileEdits = hasEdits(transactionParams, ['about', 'handle', 'avatarURI', 'name'])
+  const hasAccountsEdits = hasEdits(transactionParams, ['rootAccount', 'controllerAccount'])
+
+  const transactions: SubmittableExtrinsic<'rxjs'>[] = []
+
+  if (!api || (!hasProfileEdits && !hasAccountsEdits)) {
```

Comment:
```suggestion
  if (!api || !(hasProfileEdits || hasAccountsEdits)) {
```
Not that it's ultimately better, but for me personally this reads easier.